app/assets/javascripts/models/beta/story.js

```diff
@@ -325,6 +325,34 @@ export const cloneStory = (story) => {
   };
 };
 
+export const mergeWithFetchedStories = (
+  currentStories,
+  fetchedStories,
+  PastStoryIds
+) => {
+  const storyNotSaved = currentStories.filter(isNew);
+
+  const editedStories = fetchedStories.map((fetchedStory) => {
+    const existingStory = currentStories.find(
+      (story) => story.id === fetchedStory.id
+    );
+    if (existingStory && !existingStory.collapsed) {
+      return {
+        ...fetchedStory,
+        collapsed: false,
+        _editing: existingStory._editing,
+      };
+    }
+    return fetchedStory;
+  });
+
+  const pastStories = PastStoryIds.map((id) =>
+    currentStories.find((story) => story.id === id)
+  ).filter((story) => story !== undefined);
```

Comment:
Can this be converted to a `reduce` so we avoid two nested loops?

Reply:
But the idea of the reduce() is to return just a single value, in this case I want to return an array of objects. I don't know how reduce() can help me to reduce time complexity from O(n²) to O(n).